content/Improve your Rails App searches with Elasticsearch - Leandro Dominguez.md

```diff
@@ -0,0 +1,214 @@
+
+In this tutorial we will assume you have a basic understanding of Ruby on Rails applications and have Elasticsearch server installed and a Rails app we will integrate with. If is not your case, you can [follow the official Elastic documentation]([https://www.elastic.co/guide/en/elasticsearch/reference/current/brew.html](https://www.elastic.co/guide/en/elasticsearch/reference/current/brew.html)) and use [our awesome Ruby on Rails boilerplate]([https://github.com/rootstrap/rails_api_base](https://github.com/rootstrap/rails_api_base)).
+
+## But first… What is Elasticsearch?
+
+In their [official web]([https://www.elastic.co/elasticsearch/](https://www.elastic.co/elasticsearch/)) they describe Elasticsearch as a “_distributed, RESTful search and analytics engine capable of addressing a growing number of use cases. (...) it centrally stores your data for lightning fast search, fine‑tuned relevancy, and powerful analytics that scale with ease”_ and in my humble opinion that’s pretty true due to its scalable and distributed architecture.
+
+This [open source project]([https://github.com/elastic/elasticsearch](https://github.com/elastic/elasticsearch)) was built under the full-featured text search engine [Apache Lucene]([https://lucene.apache.org/core/](https://lucene.apache.org/core/)) and there is where most of the efficient magic happens.
+
+In a nutshell, we’ll use Elasticsearch as a secondary database (a non-relational one), taking advantage of its text pre-processing and indexing power.
+
+## A small disclaimer
+
+Although there are lots of interesting and useful gems in Rails to work with Elasticsearch like [chewy](https://github.com/toptal/chewy), [searchkick](https://github.com/ankane/searchkick) or [search_flip](https://github.com/mrkamel/search_flip) (which I encourage you to take a look at), in this project we are going to use the two basic libraries provided by the Elasticsearch team: [elasticsearch-rails](https://github.com/elastic/elasticsearch-rails) and [elasticsearch-model](https://github.com/elastic/elasticsearch-rails/tree/master/elasticsearch-model). These two gems will let us both interact with the Elastic server instance using an ActiveRecord like syntax without having to write all the specific DSL and also update our records in the Elastic instance so we always keep as synchronized as possible with our relational database.
+
+## Let’s get down to work!
+
+### What are we going to build?
+
+We’re gonna build a really simple microblogging searcher, where we are going to be available to find posts accepting possible typos and highlighting the matching results really fast and performant. For this, we will use text analyzers and tokenizers ([natural language processing tools]([https://en.wikipedia.org/wiki/Natural_language_processing](https://en.wikipedia.org/wiki/Natural_language_processing))) that come out the box with Elastic.
```

Comment:
`where we are going to make available...`?

`performantly` instead of `performant`

`For this, we will use text analyzers and tokenizers ([natural language processing tools](....` is there an `and` missing here?

Reply:
![Screen Shot 2020-07-28 at 11 55 31 AM](https://user-images.githubusercontent.com/42643389/88682620-3f772c00-d0c9-11ea-9467-c8a0e995592b.png)
No, is just a parenthesis clarifying what text analyzers and tokenizers are. Is everything ok here?

Reply:
ahhh got it! nevermind then